src/plugins/linkAction/LinkAction.js

```diff
@@ -0,0 +1,117 @@
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+export default class LinkAction {
+    constructor(openmct) {
+        this.name = 'Create Link';
+        this.key = 'link';
+        this.description = 'Create Link to object in another location.';
+        this.cssClass = "icon-link";
+        this.group = "action";
+        this.priority = 7;
+
+        this.openmct = openmct;
+    }
+
+    async invoke(objectPath) {
+        let objectToLink = objectPath[0];
+        let dialogService = this.openmct.$injector.get('dialogService');
+        let dialogForm = this.getDialogForm(objectToLink);
+        let userInput = await dialogService.getUserInput(dialogForm, {});
+        let newParent = userInput.location;
+
+        // legacy check
+        if (this.isLegacyDomainObject(newParent)) {
+            newParent = await this.convertFromLegacy(newParent);
+        }
+
+        this.linkInNewParent(objectToLink, newParent);
+    }
+
+    isLegacyDomainObject(domainObject) {
+        return domainObject.getCapability !== undefined;
+    }
+
+    async convertFromLegacy(legacyDomainObject) {
+        let objectContext = legacyDomainObject.getCapability('context');
+        let domainObject = await this.openmct.objects.get(objectContext.domainObject.id);
+
+        return domainObject;
+    }
+
+    linkInNewParent(child, newParent) {
+        let compositionCollection = this.openmct.composition.get(newParent);
+
+        compositionCollection.add(child);
+    }
+
+    getDialogForm(objectToLink) {
+        let validate = this.validate(objectToLink);
+
+        return {
+            name: `Link "${objectToLink.name}" to a New Location`,
+            sections: [
+                {
+                    rows: [
+                        {
+                            name: "Link To",
+                            control: "locator",
+                            validate,
+                            key: 'location'
+                        }
+                    ]
+                }
+            ]
+        };
+    }
+
+    validate(objectToLink) {
+        return (parentObject) => {
+            let parentCandidateKeystring = this.openmct.objects.makeKeyString(parentObject.getId());
+            let objectToLinkKeystring = this.openmct.objects.makeKeyString(objectToLink.identifier);
+            let sameObjectOrChildAlready = parentCandidateKeystring === objectToLinkKeystring
+                || parentObject.getModel().composition.includes(objectToLinkKeystring);
+
+            // the same object or a child already, not valid
+            if (sameObjectOrChildAlready) {
+                return false;
+            }
+
+            if (parentObject.getModel().locked) {
+                return false;
+            }
+
+            // can contain
+            return this.openmct.composition.checkPolicy(
```

Comment:
Can we get rid of this legacy code and use a new Composition policy instead?
See `ConditionSetCompositionPolicy`